frameworklauncher/src/main/java/com/microsoft/frameworklauncher/applicationmaster/GpuAllocationManager.java

```diff
@@ -73,14 +77,28 @@ public synchronized Node allocateCandidateRequestNode(ResourceDescriptor request
 
       if (nodeLabel != null) {
         Set<String> nodeLabels = entry.getValue().getNodeLabels();
-        if (!nodeLabels.contains(nodeLabel)) {
-          LOGGER.logInfo(
+        if (nodeLabels != null && nodeLabels.size() > 0 && !nodeLabels.contains(nodeLabel)) {
+          LOGGER.logDebug(
               "allocateCandidateRequestNode: Skip node %s, label does not match:%s",
               entry.getValue().getHostName(), nodeLabel);
           continue;
         }
       }
-
+      if (nodeGpuType != null && nodes.size() > 0) {
+        if (!nodes.containsKey(entry.getValue().getHostName())) {
+          LOGGER.logWarning(
+              "allocateCandidateRequestNode: Skip node %s, getGpuType not set",
+              entry.getValue().getHostName());
+          continue;
+        }
+        String gpuType = (nodes.get(entry.getValue().getHostName())).getGpuType();
+        if (!nodeGpuType.equals(gpuType)) {
+          LOGGER.logDebug(
+              "allocateCandidateRequestNode: Skip node %s (gpuTypeLabel:%s), labels don't match: request nodeLabel: %s",
```

Comment:
>(gpuTypeLabel:%s), labels don't match: request nodeLabel: %s", [](start = 58, length = 62)

gpuType

Reply:
remove label pls

---
In reply to: [158486901](https://github.com/Microsoft/pai/pull/128#discussion_r158486901) [](ancestors = 158486901)